### DISCUSSION TITLE
Disable processes Sec Onion 2.3

### DISCUSSION THREAD

**Discussion by frustratedlinuxadmin** [Security-Onion-Solutions/securityonion#2205] [Unsupported Versions]
Hi!

How can i disable processes in sec onion? Mainly Suricata etc.

When i stop the processes it gets autostarted again, by salt i presume?
should i edit the default.yaml file under suricata and set enabled to no on everything? Or is it a better way to do this? 

Thanks!

**Comment by weslambert** [comment]
To be clear, you are looking to permanently disable the process?

**Reply by endpointsec** [reply]
Not the OP but we are using SO2 for CSM and not NSM, so zeek and suricata could be permanently disabled. They are restarted intermittently by the salt stack I believe.

**Reply by frustratedlinuxadmin** [reply]
Hi! Yes i'm looking to permanently disabling them :) I'd like to keep zeek though, as i wan't the metadata generation, just not the threat detection :) 

I know this is outside of the scope though!

**Comment by frustratedlinuxadmin** [comment]
Does anyone have a clue? :)

**Reply by weslambert** [reply]
You could technically modify copy `/opt/so/saltstack/default/salt/top.sls` to `/opt/so/saltstack/local/salt`, removing the entry for the service you wish to disable, however this is not recommended, as it may cause issues later because you won't receive updates to the file (the `local` file).  We are working on making this easier in a future release.  You can already do this with `steno` in the pillar, as well as some other services:

Ex:
https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/top.sls#L2-L16
